packages/nouns-contracts/test/foundry/StreamEscrow.t.sol

```diff
@@ -0,0 +1,718 @@
+
+        // cancel stream 2
+        vm.prank(user2);
+        nounsToken.approve(address(escrow), 2);
+        vm.prank(user2);
+        escrow.cancelStream(2);
+
+        // check user2 refund amount
+        // stream 2 has streamed for 51 days, 51 * 0.02 = 1.02 eth, 2 eth - 1.02 eth = 0.98 eth
+        assertEq(user2.balance, 0.98 ether);
+
+        // forward 50 days
+        for (uint i; i < 50; i++) {
+            forwardOneDay();
+        }
+        // total streamed = 6 - 0.98 = 5.02 eth
+        assertEq(ethRecipient.balance, 5.02 ether);
+    }
+}
+
+contract DAOSettersTest is BaseStreamEscrowTest {
+    function test_setDAOExecutorAddress_onlyDAO() public {
+        vm.expectRevert('only dao');
+        escrow.setDAOExecutorAddress(address(1));
+    }
+
+    function test_setDAOExecutorAddress() public {
+        vm.prank(treasury);
+        vm.expectEmit();
+        emit StreamEscrow.DAOExecutorAddressSet(address(1));
+        escrow.setDAOExecutorAddress(address(1));
+
+        assertEq(escrow.daoExecutor(), address(1));
+
+        // treasury can't call setter now
+        vm.prank(treasury);
+        vm.expectRevert('only dao');
+        escrow.setDAOExecutorAddress(address(2));
+
+        // address(1) needs to call it
+        vm.prank(address(1));
+        escrow.setDAOExecutorAddress(address(2));
+    }
+
+    function test_setETHRecipient_onlyDAO() public {
+        vm.expectRevert('only dao');
+        escrow.setETHRecipient(address(1));
+    }
+
+    function test_setETHRecipient() public {
+        vm.prank(treasury);
+        vm.expectEmit();
+        emit StreamEscrow.ETHRecipientSet(makeAddr('ethRecipient2'));
+        escrow.setETHRecipient(makeAddr('ethRecipient2'));
+
+        // create a stream
+        vm.prank(streamCreator);
+        escrow.forwardAllAndCreateStream{ value: 1 ether }({ nounId: 1, streamLengthInTicks: 100 });
+        forwardOneDay();
+
+        // test that ethRecipient2 received the eth
+        assertEq(makeAddr('ethRecipient2').balance, 0.01 ether);
+    }
+
+    function test_setNounsRecipient_onlyDAO() public {
+        vm.expectRevert('only dao');
+        escrow.setNounsRecipient(address(1));
+    }
+
+    function test_setNounsRecipient() public {
+        vm.prank(treasury);
+        vm.expectEmit();
+        emit StreamEscrow.NounsRecipientSet(makeAddr('nounsRecipient2'));
+        escrow.setNounsRecipient(makeAddr('nounsRecipient2'));
+
+        // create a stream
+        vm.prank(streamCreator);
+        escrow.forwardAllAndCreateStream{ value: 1 ether }({ nounId: 1, streamLengthInTicks: 100 });
+
+        // cancel stream
+        vm.prank(streamCreator);
+        nounsToken.approve(address(escrow), 1);
+        vm.prank(streamCreator);
+        escrow.cancelStream(1);
+
+        // check that new recipient received the noun
+        assertEq(nounsToken.ownerOf(1), makeAddr('nounsRecipient2'));
+    }
+}
+
+contract StreamEscrowGasTest is BaseStreamEscrowTest {
```

Comment:
are you planning to write the output to some gas profiling text file?

Reply:
I just captured some values in the comment lines, feels good enough